playwright/integrations/unauthenticated-chat-with-prechat.spec.ts

```diff
@@ -57,4 +57,142 @@ test.describe('UnauthenticatedChat @UnauthenticatedChatWithPrechat', () => {
+
+                await chatSDK.initialize();
+
+                await chatSDK.startChat(optionalParams);
+
+                runtimeContext.requestId = chatSDK.requestId;
+
+                const preChatSurveyRes = await chatSDK.getPreChatSurvey();
+
+                runtimeContext.preChatSurvey = preChatSurveyRes;
+
+                await chatSDK.endChat();
+
+                return runtimeContext;
+            }, { omnichannelConfig, optionalParams })
+        ]);
+
+        const { preChatSurvey } = runtimeContext;
+        const prechatSurveyBody = JSON.parse(preChatSurvey.body[2].id);
+        const prechatSurveyResponseBody = JSON.parse(optionalParams.preChatResponse.Survey);
+        expect(prechatSurveyBody.Name).toEqual(prechatSurveyResponseBody.Name);
+    });
+
+    test('ChatSDK.getPreChatSurvey() with parseToJSON set to true should return the survey as JSON format', async ({ page }) => {
+
+        await page.goto(testPage);
+
+        const optionalParams = {
+            preChatResponse: {
+                'Type': "InputSubmit",
+                'Survey': '{"Name":"OmnichannelSurvey","IsOption":false,"Order":1,"IsRequired":false,"QuestionText":"OmnichannelSurvey"}'
+            }
+        };
+
+        const [sessionInitRequest, sessionInitResponse, runtimeContext] = await Promise.all([
+            page.waitForRequest(request => {
+                return request.url().includes(OmnichannelEndpoints.LiveChatSessionInitPath);
+            }),
+            page.waitForResponse(response => {
+                return response.url().includes(OmnichannelEndpoints.LiveChatSessionInitPath);
+            }),
+            await page.evaluate(async ({ omnichannelConfig, optionalParams }) => {
+                const { OmnichannelChatSDK_1: OmnichannelChatSDK } = window;
+                const chatSDK = new OmnichannelChatSDK.default(omnichannelConfig);
+                const runtimeContext = {};
+
+                await chatSDK.initialize();
+
+                await chatSDK.startChat(optionalParams);
+
+                runtimeContext.requestId = chatSDK.requestId;
+
+                const parseToJSON = true;
+
+                const preChatSurveyRes = await chatSDK.getPreChatSurvey(parseToJSON);
+
+                runtimeContext.preChatSurvey = preChatSurveyRes;
+
+                await chatSDK.endChat();
+
+                return runtimeContext;
+            }, { omnichannelConfig, optionalParams })
+        ]);
+
+        const { preChatSurvey } = runtimeContext;
+        const prechatSurveyBody = JSON.parse(preChatSurvey.body[2].id);
+        const prechatSurveyResponseBody = JSON.parse(optionalParams.preChatResponse.Survey);
+        expect(prechatSurveyBody.Name).toEqual(prechatSurveyResponseBody.Name);
+    });
+
+    test('ChatSDK.getPreChatSurvey() with parseToJSON set to false should return the survey as string format', async ({ page }) => {
+
+        await page.goto(testPage);
+
+        const optionalParams = {
```

Comment:
We do not need to call `chatSDK.startChat()` & `chatSDK.endChat()`. We can simply validate the prechat survey after `chatSDK.initialize()`

You simply need to validate the `typeof preChatSurvey === 'string'`